src/_P078_Eastron.ino

```diff
@@ -173,229 +187,812 @@ boolean Plugin_078(uint8_t function, struct EventStruct *event, String& string)
     }
 
     case PLUGIN_WEBFORM_LOAD:
+    {
       {
-        {
-          const __FlashStringHelper * options_model[4] = { F("SDM120C"), F("SDM220T"), F("SDM230"), F("SDM630") };
-          addFormSelector(F("Model Type"), P078_MODEL_LABEL, 4, options_model, nullptr, P078_MODEL );
+        const __FlashStringHelper *options_model[7] =
+        { F("SDM120C"), F("SDM220T"), F("SDM230"), F("SDM630"), F("SDM72D"), F("DDM18SD"), F("SDM72D V2") };
+        addFormSelector(F("Model Type"), P078_MODEL_LABEL, 7, options_model, nullptr, P078_MODEL);
+        addFormNote(F("Submit after changing the modell to update Output Configuration."));
+        // ugly conversation logic to be backwards compatible
+        // Register table is in SDM.h file under /lib/SDM_Energy_Meter
+        // Model 0 = SDM220 & SDM120CT & SDM120 (SDM220 SDM220T SDM120C SDM120CT SDM120 share the same registers)
+        // Model 1 = SDM230
+        // Model 2 = SDM72D
+        // Model 3 = DDM18SD
+        // Model 4 = SDM630
+        // Model 5 = SDM72D V2
+        // convert Models to be backwards compatible
+        // old    F("SDM120C"), F("SDM220T"), F("SDM230"), F("SDM630")
+        // new    F("SDM220 & SDM120CT & SDM120"), F("SDM230"), F("SDM72D"), F("DDM18SD"), F("SDM630"), F("SDM72D V2")
+        // result F("SDM120C"), F("SDM220T"), F("SDM230"), F("SDM630"), F("SDM72D"), F("DDM18SD"), F("SDM72D V2")
+        if ((P078_MODEL == 0)) {
+        P078_MODEL = 0;
+        }
+        if ((P078_MODEL == 1)) {
+        P078_MODEL = 0;
+        }
+        if ((P078_MODEL == 2)) {
+        P078_MODEL = 1;
+        }
+        if ((P078_MODEL == 3)) {
+        P078_MODEL = 4;
+        }
+        if ((P078_MODEL == 4)) {
+        P078_MODEL = 2;
+        }
+        if ((P078_MODEL == 5)) {
+        P078_MODEL = 3;
+        }
+        if ((P078_MODEL == 6)) {
+        P078_MODEL = 5;
         }
 
-        {
-          // In a separate scope to free memory of String array as soon as possible
-          sensorTypeHelper_webformLoad_header();
-          const __FlashStringHelper * options[P078_NR_OUTPUT_OPTIONS];
-          for (int i = 0; i < P078_NR_OUTPUT_OPTIONS; ++i) {
-            options[i] = p078_getQueryString(i);
-          }
-          for (uint8_t i = 0; i < P078_NR_OUTPUT_VALUES; ++i) {
-            const uint8_t pconfigIndex = i + P078_QUERY1_CONFIG_POS;
-            sensorTypeHelper_loadOutputSelector(event, pconfigIndex, i, P078_NR_OUTPUT_OPTIONS, options);
-          }
+      }
+      {
+        // In a separate scope to free memory of String array as soon as possible
+        sensorTypeHelper_webformLoad_header();
+        int nrOptions = 0;
+
+        switch (P078_MODEL) {
+          case 0: nrOptions = P078_NR_OUTPUT_OPTIONS_SDM220_SDM120CT_SDM120; break;
```

Comment:
Please use an enum here for the case statements too.
just 0...6 is very hard to read and easy to mess up.